tests/functional/pv/pv_encryption/test_encrypted_rbd_volume_expansion.py

```diff
@@ -0,0 +1,240 @@
+import logging
+import pytest
+
+from ocs_ci.framework.pytest_customization.marks import green_squad
+from ocs_ci.framework.testlib import (
+    ManageTest,
+    tier1,
+    skipif_ocs_version,
+    kms_config_required,
+    skipif_managed_service,
+    skipif_hci_provider_and_client,
+    skipif_disconnected_cluster,
+    skipif_proxy_cluster,
+    config,
+)
+from ocs_ci.utility.utils import TimeoutSampler
+from ocs_ci.ocs import constants
+
+
+log = logging.getLogger(__name__)
+
+# Set the arg values based on KMS provider.
+if config.ENV_DATA["KMS_PROVIDER"].lower() == constants.HPCS_KMS_PROVIDER:
+    kmsprovider = constants.HPCS_KMS_PROVIDER
+    argnames = ["kv_version", "kms_provider"]
+    argvalues = [
+        pytest.param("v1", kmsprovider),
+    ]
+else:
+    kmsprovider = constants.VAULT_KMS_PROVIDER
+    argnames = ["kv_version", "kms_provider", "use_vault_namespace"]
+    if config.ENV_DATA.get("vault_hcp"):
+        argvalues = [
+            pytest.param(
+                "v1", kmsprovider, True, marks=pytest.mark.polarion_id("OCS-5389")
+            ),
+            pytest.param(
+                "v2", kmsprovider, True, marks=pytest.mark.polarion_id("OCS-5390")
+            ),
+        ]
+    else:
+        argvalues = [
+            pytest.param(
+                "v1", kmsprovider, False, marks=pytest.mark.polarion_id("OCS-5387")
+            ),
+            pytest.param(
+                "v2", kmsprovider, False, marks=pytest.mark.polarion_id("OCS-5388")
+            ),
+        ]
+
+
+@green_squad
+@skipif_ocs_version("<4.7")
+@kms_config_required
+@skipif_managed_service
+@skipif_hci_provider_and_client
+@skipif_disconnected_cluster
+@skipif_proxy_cluster
+class TestEncryptedVolumeExpansion(ManageTest):
+    """
+    Test Encrypted Volume Expansion
+
+    """
+
+    @pytest.fixture()
+    def setup(
+        self,
+        kv_version,
+        use_vault_namespace,
+        pv_encryption_kms_setup_factory,
+    ):
+        """
+        Setup csi-kms-connection-details configmap
+
+        """
+        log.info("Setting up csi-kms-connection-details configmap")
+        self.kms = pv_encryption_kms_setup_factory(kv_version, use_vault_namespace)
+        log.info("csi-kms-connection-details setup successful")
+
+    def verify_pvc_size(self, pod_obj, expected_size):
```

Comment:
this method seems to be generic and could be used in other places. Perhaps moving it to a common utils/helpers file?

Reply:
Done